---
title: "FindTaxaCtrbt"
author: Menghan Liu
date: December 12, 2019
output:
  md_document:
    variant: markdown_github
theme: cosmo
code_folding: hide
---


```{r setup, echo = FALSE, eval=TRUE}
knitr::opts_knit$set()
dir='/Users/menghanliu/Documents/FindTaxaCtrbt'
knitr::opts_knit$set(root.dir = dir)
knitr::opts_chunk$set(message = FALSE,warning = FALSE)
```

```{r library packae, echo = FALSE, eval=TRUE}
library(dplyr)
library(kableExtra)
```

# FindTaxaCtrbt

Author: Menghan Liu  @ [twitter](https://twitter.com/menghan_liu) 

Date: December 12, 2019

This computational pipeline is designed to launch an exhaustive 



## TOC 

1. [Installation](#installation)
2. [Preparation](#preparation)
2. [Getting started](#get_start)
3. [Tutorial](#tutorial)

X. [Citation](#citation)


## Installation {#installation}

## Preparation {#preparation}

Preparation involve two steps, where step 1 is to set up a conda environment with all required dependency; step 2 is to gather data.


**Step 1. Environment setup **

This step needs to be completed for anyone who wants to use this pipeline. 

The environment required for FindTaxaCtrbt can be handled via creating a conda environment based on the ***FindTaxaCtrbt_environment.yml*** file provided, with the following code

```
conda env create -n FindTaxaCtrbt --file FindTaxaCtrbt_environment.yml # Create a new conda environment called FindTaxaCtrbt

conda activate FindTaxaCtrbt # initiate the environment
```

I created my environment named as FindTaxaCtrbt , you can name the environment however you want.


If you have the dependencies already, you can skip this step; or install the dependencies manually, 

- Diamond (v0.9.14 currently included in the *FindTaxaCtrbt_environment.yml*)
- R 
   *  R v3.3.2
   *  R packages: dplyr, tidyr
- Parallel (v0.9.14)
   
   
   
   
**Step 2. Download data**

Depending upon experimental design and research purpose, this step needs to be customized performed. 

* First, **Reference protein**: a list of protein homologs for the microbiome function/enzyme of interest
   - a **.fasta file** containging their amino acid sequences
   - a **.csv table** containing their taxonomic origin 
  
* Second, **microbiome meta'omics samples** from publicaly avaiable samples, or freshly generated
   - metagenome or metatranscriptome samples
   

## Getting started

### Prepare

```
# Initiate conda environment
conda activate FindTaxaCtrbt 

# Switch to your directory
cd /Users/menghanliu/Documents/FindTaxaCtrbt

# Build diamond index for protein reference 
diamond makedb \
   --in frc_oxc_oxdd_uniref100.faa \ # frc_oxc_oxdd_uniref100.faa as input
   --db frc_oxc_oxdd_uniref100.faa  
   
# Check whether the database is sucesfully built 
ls frc_oxc_oxdd_uniref100.faa.dmnd
```


### Run program

```
# Make a new folder to store the results 
mkdir -p output

# Run the program
bash scripts/FindTaxaCtrbt.sh  \n
     frc_oxc_oxdd_uniref100.faa \n
     data/MTG/CSM5FZ42.fastq.gz \n
     test_result test_result/
```




## Test run with sample data {#tutorial}
For example, I am interested in the oxalyl-coA decarboxylase (OXC), which is involved in the oxalate degradation function.

Therefore, I found a [homologous protein family](http://www.ebi.ac.uk/interpro/entry/InterPro/IPR017660/) for OXC at Uniprot.



```{r example table, echo=FALSE,eval=TRUE}
table_dir='/Volumes/Research/blaserlab/blaserlabspace/Members/Menghan_Liu/ODE/human_metagenomics/0_data/ODE_tnaA_info.csv'

read.csv(table_dir) %>%
  head() %>%
  dplyr::mutate(UniprotID = name,
                Strain = Species) %>%
  tidyr::separate(Species, into = c("Genus","Species"),sep=" ",extra="drop") %>%
  dplyr::select(UniprotID,Description,Phylum, Family, Genus, Species, Strain) %>%
  knitr::kable(format = "markdown") %>%
  kable_styling(font_size = 7, full_width = F) 
```

### Prepare

```
# Initiate conda environment
conda activate FindTaxaCtrbt 

# Switch to your directory
cd /Users/menghanliu/Documents/FindTaxaCtrbt

# Build diamond index for protein reference 
diamond makedb \
   --in frc_oxc_oxdd_uniref100.faa \ # frc_oxc_oxdd_uniref100.faa as input
   --db frc_oxc_oxdd_uniref100.faa  
   
# Check whether the database is sucesfully built 
ls frc_oxc_oxdd_uniref100.faa.dmnd
```


### Run program

```
# Make a new folder to store the results 
mkdir -p output

# Run the program
bash scripts/FindTaxaCtrbt.sh  \n
     frc_oxc_oxdd_uniref100.faa \n
     data/MTG/CSM5FZ42.fastq.gz \n
     test_result test_result/
```



## Citation {#citation}





