---
title: "FindTaxaCtrbt"
author: Menghan Liu
date: December 12, 2019
output:
  md_document:
    variant: markdown_github
theme: cosmo
code_folding: hide
---


```{r setup, echo = FALSE, eval=TRUE}
knitr::opts_knit$set()
dir='/Users/menghanliu/Documents/FindTaxaCtrbt'
knitr::opts_knit$set(root.dir = dir)
knitr::opts_chunk$set(message = FALSE,warning = FALSE)
```

```{r library packae, echo = FALSE, eval=TRUE}
library(dplyr)
library(kableExtra)
```

# FindTaxaCtrbt

## TOC 

1. [Installation](#installation)
2. [Getting started](#get_start)


## Installation {#installation}

## Getting started

**Step 1. Environment setup via Conda**

The environment for sucesfully running FindTaxaCtrbt is handled by creating a conda environment via the file <p style="color:darkblue">FindTaxaCtrbt_environment.yml</p>, with the following code


```
conda env create -n FindTaxaCtrbt --file FindTaxaCtrbt_environment.yml
```

we created a conda environment named FindTaxaCtrbt, and in order to initiate the environment we just simply run 

```
conda activate FindTaxaCtrbt
```

Of course, you can name the environment however you want, by just substitute the conda env name parameter `conda env create -n <your_customized_environment_name> --file FindTaxaCtrbt_environment.yml `


The dependencies include manually install dependency

- Diamond (v0.9.14 currently included in the *FindTaxaCtrbt_environment.yml*)
- R 
   *  R v3.3.2
   *  R packages: dplyr, tidyr
   
**Step 2. Download data**

Two part of input data is required 

1. a comprehensive list of protein homologs for the function of interest
   - a .fasta file containging their amino acid sequences  
   - a .csv table containing their taxonomic origin 
  
  
  i.e I am interested in the oxalate degradation function.
```{r, echo=FALSE,eval=TRUE}
table_dir='/Volumes/Research/blaserlab/blaserlabspace/Members/Menghan_Liu/ODE/human_metagenomics/0_data/ODE_tnaA_info.csv'
read.csv(table_dir) %>%
  head %>%
  kable(row.names = F)
```
  
  
2. Metagenomics and metatranscriptomics sample for the microbial community you're interested in.


**Step 3. prepare data**
  
  
  



